keras/utils/image_dataset_utils.py

```diff
@@ -410,25 +410,28 @@ def load_image(
             "Only one of pad_to_aspect_ratio, crop_to_aspect_ratio can be set to True"
         )
 
-    if data_format == "channels_first":
-        img = tf.transpose(img, (2, 0, 1))
-
-    if pad_to_aspect_ratio:
-        img = tf.image.resize_with_pad(
-            img, image_size[0], image_size[1], method=interpolation
-        )
-    elif crop_to_aspect_ratio:
+    if crop_to_aspect_ratio:
```

Comment:
Change the order of transpose to behind the pad operation. I though about extracting the "channels_first" case but I think this is clearer